libsolutil/Keccak256.cpp

```diff
@@ -69,25 +69,25 @@ static uint64_t const RC[24] = \
 #define REPEAT6(e) e e e e e e
 #define REPEAT24(e) REPEAT6(e e e e)
 #define REPEAT5(e) e e e e e
-#define FOR5(v, s, e) \
+#define FOR5(type, v, s, e) \
 	v = 0;            \
-	REPEAT5(e; v += s;)
+	REPEAT5(e; v = static_cast<type>(v + s);)
 
 /*** Keccak-f[1600] ***/
 static inline void keccakf(void* state) {
-	uint64_t* a = (uint64_t*)state;
+	auto* a = (uint64_t*)state;
```

Comment:
```suggestion
	auto* a = static_cast<uint64_t*>(state);
```
I thought for a moment that you might need `reinterpret_cast` here but looks like it's not necessary when casting from `void*`.